go/libkb/globals.go

```diff
@@ -270,6 +271,12 @@ func (g *GlobalContext) Shutdown() error {
 		}
 
 		err = epick.Error()
+
+		// Set the exit code for the caller. Don't overwrite
+		// if it is set to something besides ExitCode_OK.
+		if g.ExitCode == keybase1.ExitCode_OK && err != nil {
```

Comment:
These lines are hosing you.  Remove.  Don't reset the error code if there's an error on shutdown; this will fix your linux (and OSX) problems.